packages/block-library/src/table/edit.js

```diff
@@ -0,0 +1,295 @@
+		const { selectedCell } = this.state;
+
+		return [
+			{
+				icon: 'table-row-before',
+				title: __( 'Add Row Before' ),
+				isDisabled: ! selectedCell,
+				onClick: selectedCell && this.createOnInsertRow( selectedCell ),
+			},
+			{
+				icon: 'table-row-after',
+				title: __( 'Add Row After' ),
+				isDisabled: ! selectedCell,
+				onClick: selectedCell && this.createOnInsertRow( selectedCell, 1 ),
+			},
+			{
+				icon: 'table-row-delete',
+				title: __( 'Delete Row' ),
+				isDisabled: ! selectedCell,
+				onClick: selectedCell && this.createOnDeleteRow( selectedCell ),
+			},
+			{
+				icon: 'table-col-before',
+				title: __( 'Add Column Before' ),
+				isDisabled: ! selectedCell,
+				onClick: selectedCell && this.createOnInsertColumn( selectedCell ),
+			},
+			{
+				icon: 'table-col-after',
+				title: __( 'Add Column After' ),
+				isDisabled: ! selectedCell,
+				onClick: selectedCell && this.createOnInsertColumn( selectedCell, 1 ),
+			},
+			{
+				icon: 'table-col-delete',
+				title: __( 'Delete Column' ),
+				isDisabled: ! selectedCell,
+				onClick: selectedCell && this.createOnDeleteColumn( selectedCell ),
+			},
+		];
+	}
+
+	renderSection( { type, rows } ) {
+		if ( ! rows.length ) {
+			return null;
+		}
+
+		const Tag = `t${ type }`;
+		const { selectedCell } = this.state;
+
+		return (
+			<Tag>
+				{ rows.map( ( { cells }, rowIndex ) =>
+					<tr key={ rowIndex }>
+						{ cells.map( ( { content, tag: CellTag }, columnIndex ) => {
+							const isSelected = selectedCell && (
+								type === selectedCell.section &&
+								rowIndex === selectedCell.rowIndex &&
+								columnIndex === selectedCell.columnIndex
+							);
+
+							const id = {
+								section: type,
+								rowIndex,
+								columnIndex,
+							};
+
+							const classes = classnames( {
+								'is-selected': isSelected,
+							} );
+
+							return (
+								<CellTag key={ columnIndex } className={ classes }>
+									<RichText
+										value={ content }
+										onChange={ this.createOnChange( id ) }
+										placeholder={ __( 'Add cell content' ) }
+										unstableOnFocus={ this.createOnFocus( id ) }
+									/>
+								</CellTag>
+							);
+						} ) }
+					</tr>
+				) }
+			</Tag>
+		);
+	}
+
+	componentDidUpdate() {
+		const { isSelected } = this.props;
+		const { selectedCell } = this.state;
+
+		if ( ! isSelected && selectedCell ) {
+			this.setState( { selectedCell: null } );
+		}
+	}
+
+	render() {
+		const { attributes, className } = this.props;
+		const { initialRowCount, initialColumnCount } = this.state;
+		const { hasFixedLayout, head, body, foot } = attributes;
+		const isEmpty = ! head.length && ! body.length && ! foot.length;
+		const Section = this.renderSection;
+
+		if ( isEmpty ) {
+			return (
+				<Fragment>
+					<TextControl
+						type="number"
+						label={ __( 'Column Count' ) }
+						value={ initialColumnCount }
+						onChange={ this.onChangeInitialColumnCount }
+					/>
```

Comment:
I think it would feel nice to auto-select this field when the table block is inserted.

Reply:
Getting the following error when adding `autoFocus`:

```
 error  The autoFocus prop should not be used, as it can reduce usability and accessibility for users
```

Should this be something that is part of writing flow instead, that is, always focussing content when a block is created?

Reply:
Oh. Sure, sounds good 🙂